src/scripts/component-preview-iframe.js

```diff
@@ -0,0 +1,99 @@
+// Observe size of <body> in iframe and send height to parent to allow proper frame size when showcasing component.
+setTimeout(() => {
+
+  // Resize observer to send window width to parent
+  const resize = new ResizeObserver(element => {
+    const cr = element[0].contentRect;
+
+    window.parent.postMessage({ height: `${cr.height + 48}` }, '*');
+  });
+
+  resize.observe(document.body);
+
+  // Extra logic for elements that use position: absolute; styling
+  // Use MutationObservers to change body height to send proper height to documentation page
+
+  // MutationObserver to handle displaying elements that don't increase the window size
+  // For: theme and topic menu and top-nav dropdown
+  const handleMutations = mutationsList => {
+    for (const mutation of mutationsList) {
+      if (mutation.target.nodeName == 'BUTTON') {
+        if (mutation.target.ariaExpanded === 'true') {
+          document.body.style = 'height: 1000px';
+        } else {
+          document.body.style = 'height: auto;';
+        }
+      } else if (mutation.target.nodeName == 'GCDS-NAV-GROUP') {
+        if (
+          mutation.target.getAttribute(mutation.attributeName) != null &&
+          window.innerWidth >= 1024
+        ) {
+          document.body.style = 'height: 500px;';
+        } else {
+          document.body.style = 'height: auto;';
```

Comment:
Instead of all the CSS inline styles here, could we use CSS classes to style the height?